### PR TITLE
Update vendor dependencies

### PR DESCRIPTION
**Pull request by cynepco3hahue**
- Align k8s.io dependencies version to k8s-1.11.2
- Get rid of kubernetes-incubator/apiserver-builder dependency
  This package depends on release-1.9 or on release-1.12, in both
  cases it does not work with our dependencies

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -27,37 +27,36 @@ required = [
...
 
+[[override]]
+  name = "k8s.io/kube-aggregator"
+  version = "kubernetes-1.11.1"
+
 [[override]]
   name = "k8s.io/code-generator"
-  # revision for tag "kubernetes-1.11.2"
-  revision = "6702109cc68eb6fe6350b83e14407c8d7309fd1a"
+  version = "kubernetes-1.11.1"
```
Comment (ingvagabund) [inline Gopkg.toml]: Why not `1.11.2`?

File: test/integration/main.go [hunk 2]
```diff
@@ -58,7 +59,7 @@ type TestConfig struct {
 
 // NewTestConfig creates new test config with clients
 func NewTestConfig(kubeconfig string) *TestConfig {
-	config, err := controller.GetConfig(kubeconfig)
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
```
Comment (ingvagabund) [inline test/integration/main.go]: +1

File: Gopkg.toml (hunk 1 above)
Comment (cynepco3hahue) [reply Gopkg.toml]: I checked other openshift repositories that we depend on like `github.com/openshift/api`, `github.com/openshift/client-go` and they use `kubernetes-1.11.1`

File: test/integration/manifests/machineset.yaml [hunk 3]
```diff
@@ -21,7 +21,7 @@ spec:
         sigs.k8s.io/cluster-api-machine-type: worker
         sigs.k8s.io/cluster-api-machineset: worker
     spec:
-      providerConfig:
+      providerSpec:
```
Comment (ingvagabund) [inline test/integration/manifests/machineset.yaml]: Ha, this got forgotten. Please, remove this from the PR. The field can be renamed once we remove the `.providerConfig` field completely.

File: Gopkg.toml (hunk 1 above)
Comment (ingvagabund) [reply Gopkg.toml]: fair enough. Hopefully there is no signification difference (e.g. known errors in 1.11.1 fixed in 1.11.2).

Comment (cynepco3hahue) [reply Gopkg.toml]: @ingvagabund, in the end, I updated dependencies to 1.11.2 because `controller-runtime` depends on it

**Comment by cynepco3hahue** [comment]
@ingvagabund hi, do you know how can I update actuator images with the latest changes?
I can see that https://hub.docker.com/r/openshift/origin-aws-machine-controllers update 19 days ago, so it definitely does not include `providerSpec` changes.

**Comment by ingvagabund** [comment]
@cynepco3hahue yeah, the job responsible for building the docker images is most likely broken/not running. @paulfantom can you take a look at it?

**Comment by ingvagabund** [comment]
@cynepco3hahue can you break the commit into two? One that updates deps and one that changes the mao code? It's easier to review then. Plus, the PR summary does not reflect what the PR does. Can you more elaborate on that? E.g. `Remove kubernetes-incubator/apiserver-builder dependency`

**Comment by cynepco3hahue** [comment]
@ingvagabund I separated change where I drop `kubernetes-incubator/apiserver-builder` dependency under separate PR https://github.com/openshift/machine-api-operator/pull/169 and will rebase this one after it.

**Comment by paulfantom** [comment]
@cynepco3hahue we have images on quay, for this repo image is stored in https://quay.io/repository/openshift/origin-machine-api-operator

**Comment by cynepco3hahue** [comment]
@paulfantom Do I need to update this repo to take images from `quay` and from `docker`?

**Comment by paulfantom** [comment]
Yes, we should use quay where possible.

**Comment by cynepco3hahue** [comment]
@paulfantom Can you point me to related images under the quay.io? I just tried to search for `machine-api-operator` and `cluster-api-provider-aws` and failed to find it.

**Comment by paulfantom** [comment]
Cloud team images are here:
https://quay.io/repository/openshift/origin-libvirt-machine-controllers
https://quay.io/repository/openshift/origin-aws-machine-controllers
https://quay.io/repository/openshift/origin-machine-api-operator

**Comment by ingvagabund** [comment]
@cynepco3hahue can you rebase?

**Comment by cynepco3hahue** [comment]
@ingvagabund Sure and sorry for the delay, I had some snowboard PTO :smile:

**Comment by ingvagabund** [comment]
> @ingvagabund Sure and sorry for the delay, I had some snowboard PTO :smile: 

Take your time :) Not urgent, relaxing and having good time is better :P

File: Gopkg.lock [hunk 4]
```diff
@@ -285,6 +279,14 @@
   revision = "1624edc4454b8682399def8740d46db5e4362ba4"
   version = "v1.1.5"
 
+[[projects]]
```
Comment (enxebre) [inline Gopkg.lock]: why do we need this? How did it end up in the lock file?

File: pkg/operator/status.go [hunk 5]
```diff
@@ -103,7 +103,12 @@ func (optr *Operator) syncStatus(conds []osconfigv1.ClusterOperatorStatusConditi
 			Name:      optr.name,
 		},
 		Status: osconfigv1.ClusterOperatorStatus{
-			Version: version.Raw,
```
Comment (enxebre) [inline pkg/operator/status.go]: This belongs to a different commit. This effort it's being tracked here https://jira.coreos.com/browse/CLOUD-332

File: Gopkg.lock (hunk 4 above)
Comment (cynepco3hahue) [reply Gopkg.lock]: Hm, I just ran `dep ensure -v`, I will check the dependency tree to see from where it is coming.

File: pkg/operator/status.go (hunk 5 above)
Comment (cynepco3hahue) [reply pkg/operator/status.go]: I need to update it here because of `dep` updates to the latest changes, at least from CI I can see that it does not break anything.

File: Gopkg.lock (hunk 4 above)
Comment (cynepco3hahue) [reply Gopkg.lock]: Ok, `logrus` v1.3.0 has this dependency

Comment (enxebre) [reply Gopkg.lock]: definitely sg

**Comment by cynepco3hahue** [comment]
After the pivoting of cluster API this PR does not relevant anymore.
